### PR TITLE
Convert filters list to groovy script list

### PR DESCRIPTION
**Pull request by FranckLecuyer**
**Branch based upon 'saveNewList' branch**

### CODE REVIEW

**Comment by FranckLecuyer** [comment]
to be merged after PR: https://github.com/gridsuite/gridactions-app/pull/30

File: src/utils/rest-api.js [hunk 1]
```diff
@@ -133,6 +133,63 @@ export function addScriptContingencyList(name, script) {
...
+) {
+    const url =
+        PREFIX_ACTIONS_QUERIES +
+        '/v1/filters-contingency-lists/' +
+        encodeURIComponent(name) +
+        '/replace-with-script';
+    return backendFetch(url, {
+        method: 'put',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
```
Comment (geofjamg) [inline src/utils/rest-api.js]: Why not just give the name of the filter list, everything is available in the DB?

File: src/utils/rest-api.js [hunk 2]
```diff
@@ -133,6 +133,63 @@ export function addScriptContingencyList(name, script) {
...
+export function newScriptFromFiltersContingencyList(name, filtersContingency) {
+    const url =
+        PREFIX_ACTIONS_QUERIES +
+        '/v1/filters-contingency-lists/' +
+        encodeURIComponent(name) +
+        '/new-script';
+    return backendFetch(url, {
+        method: 'put',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
```
Comment (geofjamg) [inline src/utils/rest-api.js]: Same

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: src/utils/rest-api.js (hunk 1 above)
Comment (FranckLecuyer) [reply src/utils/rest-api.js]: I dit it that way at first, to handle both cases (filter list with or without modifications in progress), but with the recent refactoring of code done in gridactions, the 2 methods 'replaceFiltersWithScriptContingencyList' and 'newScriptFromFiltersContingencyList' should effectively now use 'newFiltersContingency' state as argument, and not 'currentFiltersContingency'

So, can i just do that modification ?

Comment (geofjamg) [reply src/utils/rest-api.js]: We should allow to create the script only from saved/validated filter list so, we can do it just with server data.

Comment (FranckLecuyer) [reply src/utils/rest-api.js]: So, i have to make another PR in actions-server to do that

Comment (FranckLecuyer) [reply src/utils/rest-api.js]: Done

File: src/utils/rest-api.js (hunk 2 above)
Comment (FranckLecuyer) [reply src/utils/rest-api.js]: Done

**Review by geofjamg — APPROVED** [review]

**Pull request merged**